csrc/root_domain_map.cpp

```diff
@@ -171,8 +205,6 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
     // 2. IDs that may have different extents (e.g., non indexed
     //  domains of torch_gather)
     // 3. Squeeze and unsqueeze
-    // 4. Broadcast and non broadcast
-    // 5. Symbolic ID with different extent from other ID
```

Comment:
Why are these being removed? Doesn't this cause breakage?

Reply:
I moved them within `updatePairwiseDomainMap` since that will be a common check for any special cases. The other 3 conditions were op-specific (select, torch gather etc).

Reply:
Ah ok of course. I missed that this is called in those cases now too.